docs/resources/server/advancements.md

```diff
@@ -1,173 +1,281 @@
+A `SimpleCriterionTrigger.SimpleInstance` represents a single criterion defined in the `criteria` object. Trigger instances are responsible for holding the defined conditions, and returning whether the inputs match the condition.
 
-Conditions are usually passed in through the constructor. The `SimpleCriterionTrigger.SimpleInstance` interface requires only one function, called `#player`, which returns the conditions the player must meet as an `Optional<ContextAwarePredicate>`. If the subclass is a Java record with a `player` parameter of this type (as below), the automatically generated `#player` method will suffice.
+Conditions are usually passed in through the constructor. The `SimpleCriterionTrigger.SimpleInstance` interface requires only one function, called `#player`, which returns the conditions the player must meet as an `Optional<ContextAwarePredicate>`. If the subclass is a record with a `player` parameter of this type (as below), the automatically generated `#player` method will suffice.
 
 ```java
-public record ExampleTriggerInstance(Optional<ContextAwarePredicate> player, ItemPredicate item) implements SimpleCriterionTrigger.SimpleInstance {
-    // extra methods here
-}
+public record ExampleTriggerInstance(Optional<ContextAwarePredicate> player/*, other parameters here*/)
+        implements SimpleCriterionTrigger.SimpleInstance {}
 ```
 
-:::note
 Typically, trigger instances have static helper methods which construct the full `Criterion<T>` object from the arguments to the instance. This allows these instances to be easily created during data generation, but are optional.
 
 ```java
-// In this example, EXAMPLE_TRIGGER is a DeferredHolder<CriterionTrigger<?>, ExampleTrigger>
+// In this example, EXAMPLE_TRIGGER is a DeferredHolder<CriterionTrigger<?>, ExampleTrigger>.
+// See below for how to register triggers.
 public static Criterion<ExampleTriggerInstance> instance(ContextAwarePredicate player, ItemPredicate item) {
     return EXAMPLE_TRIGGER.get().createCriterion(new ExampleTriggerInstance(Optional.of(player), item));
 }
 ```
-:::
 
 Finally, a method should be added which takes in the current data state and returns whether the user has met the necessary conditions. The conditions of the player are already checked through `SimpleCriterionTrigger#trigger(ServerPlayer, Predicate)`. Most trigger instances call this method `#matches`.
 
 ```java
-// This method is unique for each instance and is as such not overridden
-public boolean matches(ItemStack stack) {
-    // Since ItemPredicate matches a stack, a stack is the input
-    return this.item.test(stack);
+// Let's assume we have an additional ItemPredicate parameter. This can be whatever you need.
+// For example, this could also be a Predicate<LivingEntity>.
+public record ExampleTriggerInstance(Optional<ContextAwarePredicate> player, ItemPredicate predicate)
```

Comment:
I disagree, the example with `ItemPredicate` is an example and can be adapted for any predicate, therefore I think it makes more sense to use `predicate` as the name.

Reply:
I guess? It just seems weird since the `ContextAwarePredicate` is supposed to represent the `player` passed in. So looking at `predicate`, I don't necessarily understand what this is supposed to represent. But, that's more arguing naming than anything, so I'm willing to concede.